src/model/entities/segment.ts

```diff
@@ -54,6 +54,9 @@ export class Segment {
   }
 
   public setAsNext(): void {
+    if (!this.isSegmentOnAir) {
+      this.reset()
+    }
     this.isSegmentNext = true
     if (!this.isSegmentOnAir) {
       this.reset()
```

Comment:
I think you should take a closer look at this code ;)

Reply:
One could even say you should do a double take :D

Reply:
The reset below is from develop and not this branch. I have merged develop into the branch and removed the double reset.